third_party/txt/src/txt/platform_mac.mm

```diff
@@ -42,34 +42,76 @@
   return mgr;
 }
 
+CTFontRef MatchSystemUIFont(float desired_weight, float size) {
+  CTFontRef ct_font(
+      CTFontCreateUIFontForLanguage(kCTFontUIFontSystem, size, nullptr));
+
+  if (desired_weight == kNormalWeightValue) {
+    return ct_font;
+  }
+
+  CFMutableDictionaryRef variations(CFDictionaryCreateMutable(
+      kCFAllocatorDefault, 1, &kCFTypeDictionaryKeyCallBacks,
+      &kCFTypeDictionaryValueCallBacks));
+
+  auto add_axis_to_variations = [&variations](const FourCharCode tag,
+                                              float desired_value,
+                                              float normal_value) {
+    if (desired_value != normal_value) {
+      CFNumberRef tag_number(
+          CFNumberCreate(kCFAllocatorDefault, kCFNumberIntType, &tag));
+      CFNumberRef value_number(CFNumberCreate(
+          kCFAllocatorDefault, kCFNumberFloatType, &desired_value));
+      CFDictionarySetValue(variations, tag_number, value_number);
+    }
+  };
+  add_axis_to_variations(kWeightTag, desired_weight, kNormalWeightValue);
+
+  CFMutableDictionaryRef attributes(CFDictionaryCreateMutable(
+      kCFAllocatorDefault, 1, &kCFTypeDictionaryKeyCallBacks,
+      &kCFTypeDictionaryValueCallBacks));
+  CFDictionarySetValue(attributes, kCTFontVariationAttribute, variations);
+
+  CTFontDescriptorRef var_font_desc(
+      CTFontDescriptorCreateWithAttributes(attributes));
+
+  return CTFontCreateCopyWithAttributes(ct_font, size, nullptr, var_font_desc);
+}
+
```

Comment:
I was profiling allocations - and this seems to be leaking the `CFNumber`s, `CFMutableDictionary` and `CTFontDescriptor`. I don't see CFRelease calls anywhere? I don't think there's a `ScopedCFTypeRef` in fml so this will need to be released manually.

Reply:
@knopp could you file an issue for that so it doesn't get lost?

Reply:
I think this has been already fixed when clang tidy started complaining about missing CFRelease.

Reply:
Do you know which release will contain the memory leak fix?